Jenkinsfile

```
@@ -347,38 +374,71 @@ pipeline {
           stages {
             stage('Winlogbeat oss'){
               steps {
-                withBeatsEnv(){
-                  makeTarget("Winlogbeat oss crosscompile", "-C winlogbeat crosscompile")
+                makeTarget("Winlogbeat oss crosscompile", "-C winlogbeat crosscompile")
+              }
+            }
+            stage('Winlogbeat Windows'){
+              agent { label 'windows-immutable && windows-2019' }
+              options { skipDefaultCheckout() }
+              when {
+                beforeAgent true
+                expression {
+                  return params.windowsTest
                 }
               }
+              steps {
+                mageTargetWin("Winlogbeat Windows Unit test", "-d winlogbeat goUnitTest")
+              }
             }
           }
         }
+        stage('Winlogbeat Windows x-pack'){
+          agent { label 'windows-immutable && windows-2019' }
+          options { skipDefaultCheckout() }
+          when {
+            beforeAgent true
+            expression {
+              return params.windowsTest && env.BUILD_WINLOGBEAT_XPACK != "false"
+            }
+          }
+          steps {
+            mageTargetWin("Winlogbeat Windows Unit test", "-d x-pack/winlogbeat goUnitTest")
```

```suggestion
            mageTargetWin("Winlogbeat Windows Unit test", "-d x-pack/winlogbeat  update:fields goUnitTest")
```

That should cause the fields.yml to be generated.